hexrd/ui/hexrd_config.py

```diff
@@ -710,15 +710,29 @@ def masked_images_dict(self):
         finally:
             self.masks = prev_masks
 
+        masks_dict = {}
         images_dict = self.images_dict
         for name, img in images_dict.items():
+            final_mask = np.ones(img.shape, dtype=bool)
             for mask_name, data in masks.items():
                 if mask_name not in self.visible_masks:
                     continue
 
                 for det, mask in data:
                     if det == name:
-                        img[~mask] = 0
+                        final_mask = np.logical_and(final_mask, mask)
+            masks_dict[name] = final_mask
+
+        return masks_dict
+
+    @property
+    def masked_images_dict(self):
+        """Get an images dict where masks have been applied"""
+        images_dict = self.images_dict
+        for name, img in images_dict.items():
+            for det, mask in self.masks_dict.items():
```

Comment:
Can we set `masks_dict = self.masks_dict` outside of the loops so it doesn't rebuild the masks dict for every iteration of the images dict?

Reply:
Yes! Fixed.